es-vue-base/src/lib-components/EsFileThumbnail.vue

```diff
@@ -0,0 +1,341 @@
+                    <b-link
+                        v-if="!loading"
+                        aria-label="close-file-desktop"
+                        class="text-gray-800 text-decoration-none icon-button"
+                        @click="$emit('removeFile')">
+                        <icon-circle-x />
+                        <div class="svg-fill-wrapper bg-white" />
+                    </b-link>
+                </div>
+
+                <b-link
+                    v-if="mimeType.includes('image')"
+                    class="text-decoration-none text-gray-800"
+                    aria-label="show-preview-desktop-image"
+                    :disabled="loading"
+                    @click="$emit('showPreview')">
+                    <div
+                        :class="{
+                            'card': true,
+                            'thumbnail-inner-wrapper-x': !loading,
+                            'thumbnail-inner-wrapper': loading,
+                        }">
+                        <template v-if="!loading">
+                            <div class="h-100 w-100">
+                                <b-img
+                                    fluid
+                                    class="image-preview"
+                                    :src="fileSource"
+                                    :alt="fileName" />
+                            </div>
+                        </template>
+                        <template v-else>
+                            <div class="h-100 d-flex align-items-center justify-content-center">
+                                <es-progress-circle
+                                    :value="percentLoaded"
+                                    circle
+                                    height="65px" />
+                            </div>
+                        </template>
+                    </div>
+                </b-link>
+                <b-link
+                    v-if="mimeType.includes('application')"
+                    class="text-decoration-none text-gray-800"
+                    aria-label="show-preview-desktop-file"
+                    :disabled="loading"
+                    :href="fileSource"
+                    target="_blank">
+                    <div
+                        :class="{
+                            'card': true,
+                            'thumbnail-inner-wrapper-x': !loading,
+                            'thumbnail-inner-wrapper': loading,
+                        }">
+                        <template v-if="!loading">
+                            <div
+                                class="h-100 d-flex align-items-center justify-content-center">
+                                <icon-file-pdf v-if="mimeType.includes('pdf')" />
+                                <icon-file-doc v-if="mimeType.includes('doc') && !(mimeType.includes('docx'))" />
+                                <icon-file-docx v-if="mimeType.includes('docx')" />
```

Comment:
We should probably pick one of these to be the default if the file type matches none of our definitions. `v-else` on `doc` feels good.

Reply:
good call!!